pkg/api/osa/vlabs/validate.go

```diff
@@ -0,0 +1,154 @@
+package vlabs
+
+import (
+	"fmt"
+	"regexp"
+
+	"k8s.io/apimachinery/pkg/util/errors"
+)
+
+var regexRfc1123 = regexp.MustCompile(`(?i)` +
+	`^([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9])` +
+	`(\.([a-z0-9]|[a-z0-9][-a-z0-9]{0,61}[a-z0-9]))*$`)
+
+func isValidHostname(h string) bool {
+	return len(h) <= 255 && regexRfc1123.MatchString(h)
+}
+
+var regexAgentPoolName = regexp.MustCompile(`^[a-z][a-z0-9]{0,11}$`)
+
+// Validate validates an OpenShiftCluster struct
+func (oc *OpenShiftCluster) Validate() error {
+	if oc.Location == "" {
+		return fmt.Errorf("location must not be empty")
+	}
+	if oc.Name == "" {
+		return fmt.Errorf("name must not be empty")
+	}
+
+	return oc.Properties.Validate()
+}
+
+// Validate validates a Properties struct
+func (p *Properties) Validate() error {
+	switch p.ProvisioningState {
+	case "", Creating, Updating, Failed, Succeeded, Deleting, Migrating, Upgrading:
+	default:
+		return fmt.Errorf("invalid provisioningState %q", p.ProvisioningState)
+	}
+
+	if p.OpenShiftVersion == "" {
+		return fmt.Errorf("openShiftVersion must not be empty")
+	}
+
+	if p.PublicHostname == "" {
+		return fmt.Errorf("publicHostname must not be empty")
```

Comment:
Move in `isValidHostname`

Reply:
not doing this because of the change in semantic of the field